Mac App/Model/UserDefaults+WindowRestoration.swift

```diff
@@ -0,0 +1,164 @@
+        return VisionTypeDTO(fromUserDefaults: value).model
+    }
+
+    static func getSimulation() -> Simulation {
+        let value = UserDefaults.standard.integer(forKey: UserDefaults.SimulationKey)
+        return SimulationDTO(fromUserDefaults: value).model
+    }
+
+    static func setVision(_ vision: VisionType) {
+        let dto = VisionTypeDTO(fromVision: vision)
+        UserDefaults.standard.set(dto.userDefaultsValue, forKey: UserDefaults.VisionKey)
+    }
+
+    static func setSimulation(_ simulation: Simulation) {
+        let dto = SimulationDTO(fromSimulation: simulation)
+        UserDefaults.standard.set(dto.userDefaultsValue, forKey: UserDefaults.SimulationKey)
+    }
+
+}
+
+enum WindowRestoration {
+
+    static let VisionKey = "VisionType"
+    static let SimulationKey = "SimulationKey"
+
+    static func encodeRestorable(state: NSCoder, _ vision: VisionType, _ sim: Simulation) {
+        let encodedVision = Self.encode(vision)
+        let encodedSimulation = Self.encode(sim)
+        state.encode(encodedVision, forKey: Self.VisionKey)
+        state.encode(encodedSimulation, forKey: Self.SimulationKey)
+    }
+
+    static func decodeRestorable(state: NSCoder) -> (VisionType, Simulation) {
+        let vision = state.decodeInteger(forKey: Self.VisionKey)
+        let sim = state.decodeInteger(forKey: Self.SimulationKey)
+        return (Self.restore(vision: vision), Self.restore(simulation: sim))
+    }
+
+    private static func restore(vision state: Int) -> VisionType {
+        VisionTypeDTO(fromWindowState: state).model
+    }
+
+    private static func restore(simulation state: Int) -> Simulation {
+        SimulationDTO(fromWindowState: state).model
+    }
+
+    private static func encode(_ vision: VisionType) -> Int {
+        VisionTypeDTO(fromVision: vision).windowStateValue
+    }
+
+    private static func encode(_ simulation: Simulation) -> Int {
+        SimulationDTO(fromSimulation: simulation).windowStateValue
+    }
+}
+
+// MARK: - Data Transfer Utilities
+
+fileprivate extension UserDefaults {
+    static let VisionKey = "SimVisionType"
+    static let SimulationKey = "SimulationKey"
+}
+
+fileprivate enum SimulationDTO: Int {
+    case wicklineHCIRN
+    case machadoEtAl
+
+    var userDefaultsValue: Int { rawValue }
+    var windowStateValue: Int { rawValue }
+
+    init(fromUserDefaults value: Int) {
+        self = .init(rawValue: value) ?? .wicklineHCIRN
+    }
+
+    init(fromWindowState value: Int) {
+        self.init(fromUserDefaults: value)
+    }
+
+    init(fromSimulation sim: Simulation) {
+        switch sim {
+            case .wicklineHCIRN: self = .wicklineHCIRN
+            case .machadoEtAl:   self = .machadoEtAl
+        }
+    }
+
+    var model: Simulation {
+        switch self {
+            case .wicklineHCIRN: return .wicklineHCIRN
+            case .machadoEtAl:   return .machadoEtAl
+        }
+    }
+}
+
+fileprivate enum VisionTypeDTO: Int {
```

Comment:
In this case, it is a perhaps unnecessary habit of isolating persisted models from app models (i.e., the rawValue Int). It can simply use VisionType / Simulation and a `persistenceValue` so it is at least explicit what a change could disrupt.